pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -1582,13 +1588,20 @@ protected void trackMessage(MessageId messageId, int redeliveryCount) {
         }
     }
 
+    void increaseAvailablePermits(MessageImpl<?> msg) {
+        ClientCnx currentCnx = cnx();
+        ClientCnx msgCnx = msg.getCnx();
+        if (msgCnx == currentCnx) {
+            increaseAvailablePermits(currentCnx);
+        }
```

Comment:
Should we handle the `else` case? I think it's better to add some error logs to indicate the unexpected case.

Reply:
We can add a log, but should not be the error log

Same as here: https://github.com/apache/pulsar/pull/15162/files/94d27f5b600f08333b66d10324bf9ce76713dcb3#diff-f6e4c1c4091aa10525f331e48e66b29f22b9f7987755c1b4fb887e24f198bed6R1552